plugins/outputs/file/file.go

```diff
@@ -39,13 +43,24 @@ func (f *File) SetSerializer(serializer serializers.Serializer) {
 	f.serializer = serializer
 }
 
-func (f *File) Connect() error {
-	writers := []io.Writer{}
-
+func (f *File) Init() error {
+	var err error
 	if len(f.Files) == 0 {
 		f.Files = []string{"stdout"}
 	}
 
+	var options []internal.EncodingOption
+	if f.CompressionLevel >= 0 && f.CompressionAlgorithm != "" && f.CompressionAlgorithm != "identity" {
```

Comment:
Please think about this condition and compare it to my suggestion
```suggestion
	if f.CompressionAlgorithm == "" {
		f.CompressionAlgorithm = "identity"
	}
	if f.CompressionLevel >= 0  {
```

the behavior for algorithm `identity` or an empty algorithm is

| compression level | your code | my code |
| --- | ---| --- |
| not set | no option created | no option created |
| < 0 | no option created | no option created |
| >= 0 | silently ignore the set compression level | error about invalid compression level |

As you can see, my code will explicitly enforces the compression-level to be _unset_ for `identity` which is the right thing as it does nothing. If you silently ignore it, the user might have the impression that setting does something but is doesn't...